### ISSUE TITLE
Comparison with `tower_http::request_id`

### ISSUE THREAD

**Issue by samuela** [imbolc/tower-request-id#4]
What are the pros and cons of using this crate vs [`tower_http::request_id`](https://docs.rs/tower-http/latest/tower_http/request_id/index.html)?

**Comment by imbolc** [comment]
From the docs it seems that you have to implement `tower_http::request_id::SetRequestId` yourself, while this crate provides you with a ready to use middleware.

**Comment by samuela** [comment]
IIUC they also offer [`tower_http::request_id::MakeRequestUuid`](https://docs.rs/tower-http/latest/tower_http/request_id/struct.MakeRequestUuid.html) to automatically generate IDs. Not sure exactly what the ergonomics of that are, however.

**Comment by imbolc** [comment]
Yeah, seems so. I've just skimmed the docs once more, they also accept id from headers, so it can be created on a client.

I wouldn't create this crate if I knew about `tower_http::request_id` at the moment. Now it worth considering rewriting this crate by implementing their traits to ulid. Ulids still faster to generate and they're lexicographically sorted (meaning you can sort them by creation time). So I'd accept a PR if you'd like to dig into it ;)
